letsencrypt/cli.py

```diff
@@ -1139,6 +1147,26 @@ def main(cli_args=sys.argv[1:]):
 
     return args.func(args, config, plugins)
 
+def check_config_sanity(args):
+    """Validate command line options and display error message if
+    requirements are not met.
+
+    :param args: Command line options
+    :type args: :class:`argparse.Namespace`
+
+    """
+    # Domain checks
+    if args.domains is not None:
+        # Check if there's a wildcard domain
+        if any(True for d in args.domains if d.startswith("*")):
+            raise ConfigurationError("Error: Wildcard domains are not supported")
+        # Punycode
+        if any(True for d in args.domains if "xn--" in d):
+            raise ConfigurationError("Error: Punycode domains are not supported")
+        # Check for FQDN
+        fqdn = re.compile("^((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{2,6}$")
```

Comment:
http://www.mkyong.com/regular-expressions/domain-name-regular-expression-example/

It checks various points: 
* The domain name should be a-z | A-Z | 0-9 and hyphen(-)
* The domain name should between 1 and 63 characters long
* Last Tld must be at least two characters, and a maximum of 6 characters
* The domain name should not start or end with hyphen (-) (e.g. -google.com or google-.com)
* The domain name can be a subdomain (e.g. mkyong.blogspot.com)